http_urls.py

```diff
@@ -10,4 +11,5 @@
 
 urlpatterns = [
     path('', include(router.urls)),
+    path('application/<int:id>/secret_data/',ApplicationHiddenDetailView.as_view(),name="hidden-detail"),
```

Comment:
Can we instead restructure it like this - `application/secret_data/`? The benefit will be that you won't need to get the password from `POST` data and abstract `id` from the URL.

Reply:
Yes sure.
Thank you for the suggesiton.